pkg/bgpv1/agent/controller.go

```diff
@@ -103,67 +121,119 @@ type ControllerParams struct {
 // This implementation defines which BGP backend will be used (GoBGP, FRR, Bird, etc...)
 // NOTE: only GoBGP currently implemented.
 func NewController(params ControllerParams) (*Controller, error) {
-	// If the BGP control plane is disabled, just return nil. This way the hive dependency graph is always static
-	// regardless of config. The lifecycle has not been appended so no work will be done.
-	if !params.DaemonConfig.BGPControlPlaneEnabled() {
+	// If both BGP control plane versions are either enabld or disabled, just return nil.
+	// This way the hive dependency graph is always static regardless of config. The lifecycle
+	// has not been appended so no work will be done.
+	if (!params.DaemonConfig.BGPControlPlaneEnabled() && !params.DaemonConfig.BGPControlPlaneV2Enabled()) ||
```

Comment:
My thinking was only one BGP CP version should be enabled at a time until we can support conflict resolution between the versions. I will remove it since https://github.com/cilium/cilium/pull/32426 was merged.